d2bs/kolbot/libs/common/Misc.js

```diff
@@ -750,6 +791,214 @@ var Item = {
+        }
+
+        if (Config.AutoEquip && !me.getMerc()) {
+            return false;
+        }
+
+        var i,
+            tier = NTIP.GetMercTier(item),
+            color = Pickit.itemColor(item),
+            bodyLoc = this.getBodyLocMerc(item);
+
+        if (tier > 0 && bodyLoc) {
+            for (i = 0; i < bodyLoc.length; i += 1) {
+                // Low tier items shouldn't be kept if they can't be equipped
+                var oldTier = this.getEquippedItemMerc(bodyLoc[i]).tier;
+                if (tier > oldTier && (this.canEquipMerc(item) || !item.getFlag(0x10))) {
+                    print("\xFFc8AutoPlay :: New merc item: " + color + item.name + " (new: " + tier + ", old: " + oldTier + ")");
+                    return true;
+                }
+            }
+        }
+
+        return false;
+    },
+
+    // returns true if the item should be kept+logged, false if not
+    autoEquipMerc: function () {
+        if (!Config.AutoEquip || !me.getMerc()) {
+            return true;
+        }
+
+        var i, j, tier, bodyLoc, tome, gid, that, classid,
+            items = me.findItems(-1, 0);
+
+        if (!items) {
+			print("No Items");
+            return false;
+        }
+
+        function sortEq(a, b) {
+            if (Item.canEquipMerc(a) && Item.canEquipMerc(b)) {
+                return NTIP.GetMercTier(b) - NTIP.GetMercTier(a);
+            }
+
+            if (Item.canEquipMerc(a)) {
+                return -1;
+            }
+
+            if (Item.canEquipMerc(b)) {
+                return 1;
+            }
+
+            return 0;
+        }
+
+        me.cancel();
+
+        // Remove items without tier
+        for (i = 0; i < items.length; i += 1) {
+            if (NTIP.GetMercTier(items[i]) === 0) {
+                items.splice(i, 1);
+                i -= 1;
+            }
+        }
+
+        while (items.length > 0) {
+            items.sort(sortEq);
+
+            tier = NTIP.GetMercTier(items[0]);
+            bodyLoc = this.getBodyLocMerc(items[0]);
+
+            if (tier > 0 && bodyLoc) {
+                for (j = 0; j < bodyLoc.length; j += 1) {
+                    if ([3, 7].indexOf(items[0].location) > -1 && tier > this.getEquippedItemMerc(bodyLoc[j]).tier) { // khalim's will adjustment
+                        if (!items[0].getFlag(0x10)) { // unid
+                            tome = me.findItem(519, 0, 3);
+
+                            if (tome && tome.getStat(70) > 0) {
+                                if (items[0].location === 7) {
+                                    Town.openStash();
+                                }
+
+                                Town.identifyItem(items[0], tome);
+                            }
+                        }
+
+                        gid = items[0].gid;
+                        classid = items[0].classid;
+
+                        if (this.equipMerc(items[0], bodyLoc[j])) {
+							Misc.logItem("Equipped Merc", me.getItem(-1, -1, gid));
```

Comment:
run the eslint or jslint to fix any formatting issues like this

Reply:
I fix it